lib/src/resolve-symlinks.js

```diff
@@ -13,14 +13,22 @@ function resolveSymlinks(opt) {
 
   // A stat property is exposed on file objects as a (wanted) side effect
   function resolveFile(globFile, enc, callback) {
+    var stat = opt.statCache[globFile.path];
+    if (stat && (resolveSymlinks !== stat.isSymbolicLink())) {
```

Comment:
`resolveSymlinks !== stat.isSymbolicLink()` seems too clever.  I think it's more explicit `&&` between the two like below.